Sodium/Utils.swift

```diff
@@ -16,11 +16,23 @@ extension Utils {
     /**
      Tries to effectively zero bytes in `data`, even if optimizations are being applied to the code.
 
-     - Parameter data: The `Bytes` object to zero.
+     - Parameter data: The `Data` object to zero.
      */
-    public func zero(_ data: inout Bytes)  {
+    public func zero(_ data: inout Data)  {
         let count = data.count
-        sodium_memzero(&data, count)
+        data.withUnsafeMutableBytes {
+            dataPtr in sodium_memzero(dataPtr, count)
+        }
+    }
+
+    /**
+     Tries to effectively zero bytes in `data`, even if optimizations are being applied to the code.
+
+     - Parameter data: The `BytesContainer` object to zero.
+     */
+    public func zero(_ data: inout BytesContainer)  {
+        let count = data.count
+        sodium_memzero(&data.bytes, count)
```

Comment:
Does this make sense? I am not sure, how bytes are handled internally but I can imagine that `BytesRepresentable.bytes` is implemented in a way, that the bytes are copied. The implicit copy created will denn be zeroed but the original value will stay in place. The user of this API might think that something has been securely erased, when it actually was not. Correct me if I am wrong.

Reply:
Let me know if I'm off here, but did you perhaps misread the type (you mention `BytesRepresentable` here)? `data` is a `BytesContainer` (i.e. a struct which wraps the stored property `var bytes: [UInt8]`), but not a general `BytesRepresentable`. Thus this method applies to a known implementation, and the zeroing should be directly applied to the stored property `bytes` by reference?

For sure this couldn't be implemented over a general `BytesRepresentable` for the reasons above though.